port/common/omrfile.c

```diff
@@ -607,41 +607,33 @@ EsTranslateOpenFlags(int32_t flags)
 void
 omrfile_vprintf(struct OMRPortLibrary *portLibrary, intptr_t fd, const char *format, va_list args)
 {
-	char outputBuffer[256];
-	char *allocatedBuffer;
-	uint32_t numberWritten;
-	va_list copyOfArgs;
-
-	/* Attempt to write output to stack buffer */
-	COPY_VA_LIST(copyOfArgs, args);
-	numberWritten = portLibrary->str_vprintf(portLibrary, outputBuffer, sizeof(outputBuffer), format, copyOfArgs);
-
-	/* str_vprintf always null terminates, returns number characters written excluding the null terminator */
-	if (sizeof(outputBuffer) > (numberWritten + 1)) {
-		/* write out the buffer */
-		portLibrary->file_write_text(portLibrary, fd, outputBuffer, numberWritten);
-		return;
-	}
+	char localBuffer[256];
+	char *writeBuffer = NULL;
 
-	/* Either the buffer was too small, or it was the exact size.  Unfortunately can't tell the difference,
-	 * need to determine the size of the buffer (another call to str_vprintf) then print to the buffer,
-	 * a third call to str_vprintf
-	 */
-	COPY_VA_LIST(copyOfArgs, args);
+	/* What is size of buffer required ? str_vprintf(..,NULL,..) result includes the null terminator */
+	uintptr_t bufferSize = portLibrary->str_vprintf(portLibrary, NULL, 0, format, args);
```

Comment:
When the COPY_VA_LIST is brought back you will need to move the declaration of this variable up.